### ISSUE TITLE
Logger tag is missing

### ISSUE THREAD

**Issue by dakujem** [getsentry/sentry-php#1651]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

v4

### Steps to reproduce

After upgrading to v4 SDK the tag `logger` (along with its default value `php`) is missing.
This would not be anything note worthy if it was mentioned in the upgrading guide.

### Expected result

The removal of the tag-value pair `logger:php` mentioned in the migration guide.

### Actual result

The removal is not mentioned and the actual tag is no longer added by default, but needs to be explicitly added in the config options.

**Comment by cleptric** [comment]
We can certainly update the upgrade guide.
Do you use this tag for something in particular on sentry.io?

**Comment by dakujem** [comment]
@cleptric Yes, for filtering, it is the only tag that appears in the issue list:
![image](https://github.com/getsentry/sentry-php/assets/443067/5ed3dd23-7231-4e6e-a012-60a388b5b400)

**Comment by stayallive** [comment]
You can still filter using `sdk.name:sentry.php` instead of `logger:php`.

**Comment by FeBe95** [comment]
I am currently setting the `logger` tag like this:
```PHP
$sentryEvent = \Sentry\Event::createEvent();
$sentryEvent->setLogger('deprecations');
\Sentry\captureEvent($sentryEvent);
```

This works without any issues, which might be because of this PR, which shipped in 4.2.0 (Dec 19, 2023):
- #1657

**Comment by dakujem** [comment]
What is weird is that the tag appears in the list (visually) for some issues, and does not for other (while the tag is correctly set, as observable from the filter setting).
![image](https://github.com/user-attachments/assets/86b65097-e988-4040-b738-2bb837083cd0)
The one the tag appears for is a quite old issue, in fact.
